src/routes/examples/main/react.mdx

```diff
@@ -1,3 +1,250 @@
+});
+```
+
+Let's talk about what's going on in this test file.
+
+In the `beforeEach` hook we're creating a new server using the same `makeServer` function we defined earlier. We're initializing our server with a `test` environment, which optimizes Mirage to run inside of tests.
+
+In the `afterEach` hook we're calling `server.shutdown()`, which will destroy our Mirage server. This ensures that every test we write has its own isolated server and no server state is shared between any two tests.
+
+Finally, our test renders our App component and asserts that the names Bob and Alice show up on the page.
+
+While this test covers the data loading functionality of our component there's something missing. Testing components like this will maximize our confidence in them being able to take on various states. However, we also must be able to answer additional questions like: "What happens when the server has no users?", or "What does our component render if the backend 500's?"
+
+This is where Mirage shines, it has the ability to quickly create and test these scenarios, which makes answering those questions a breeze.
+
+Let's go back to our `src/server.js` file and make a small change to its definition.
+
+```diff
+// src/server.js
+
+- import { Server } from "miragejs"
++ import { Server, Model } from "miragejs"
+
+export function makeServer({ environment = "development" } = {}) {
+  let server = new Server({
+    environment,
+
++   models: {
++     user: Model
++   },
+
+    routes() {
+      this.namespace = "api"
+
+-     this.get("/users", () => {
+-       return {
+-         users: [
+-           {
+-             id: 1,
+-             name: "Bob",
+-           },
+-           {
+-             id: 2,
+-             name: "Alice",
+-          },
+-        ],
+-      }
+-    })
+
++    this.get('/users', (schema) => {
++      return schema.users.all();
++    });
+   },
+  })
+
+  return server
+}
+```
+
+With this change we're telling Mirage that `users` are a first class [model](/docs/data-layer/models) in our server. We're also changing the `/api/users` endpoint to respond with all users in the server, instead of a static list containing only Bob and Alice.
+
+Now we have the ability to test our server in various states. Let's use Mirage to create a test with a single user.
+
+```js
+// src/__tests__/App.test.js
+
+it("loads the a single user from our server", () => {
+  server.create('user', { name: 'John' });
+
+  const { getByTestId } = render(<App />);
+
+  await waitForElement(() => getByTestId("user-1"));
+
+  expect(getByTestId("user-1")).toHaveTextContent("John");
+});
+```
+
+By using `server.create()` we are taking advantage of Mirage's [factories](http://localhost:8000/docs/data-layer/factories) to define data that is specifically tailored to each test.
+
+We can also use Mirage to test error responses from our server.
+
+```js
+// src/__tests__/App.test.js
+
+import { Response } from "miragejs"
```

Comment:
Hey @ryanto, I think this is actually supposed to be `import { Response } from '@miragejs/server';`?

Reply:
Yeah...I just looked at one of my projects and I appear to have been importing base serializers and the `Response` from `'@miragejs/server'`.  Might have to check all the imports in this example.

Reply:
Thanks @samselikoff, awesome work!